### DISCUSSION TITLE
ramp_up_complex_vis

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#32] [Announcements]
# ramp_up_complex_vis

我们在 “Hello，World! 程序” 一节中创建了自己的第一个 VI，但是它过于简陋。接下来，我们将对它做一点改进，同时介绍 LabVIEW 的其它一些基本用法。了解了这些用法，才算是 LabVIEW 入门了。

https://lv.qizhen.xyz/ramp_up_complex_vis

**Comment by Digory-w** [comment]
"图标编辑器本身也是使用 LabVIEW 代码开发编写的，是一个很好的学习 LabVIEW 的范例程序。它的主 VI 位于“[labview]\resource\plugins\lv_icon.vi”。"
本来想学习一下这个的，但发现在16、24版均无法打开程序框图。查找NI官网后找到了这篇文章希望对大家有所帮助：
https://forums.ni.com/t5/Enhanced-Icon-Editor/Icon-Editor-Source-Files-for-LabVIEW-2011/td-p/3516740
（如果无法打开请注意网络环境）

**Reply by ruanqizhen** [reply]
多谢你的信息！我会在原文中也更新一下。

**Comment by ruanqizhen** [comment]
之前博客上的相关文章，搬到这里：

### 图标编辑器上的鼠标双击技巧

图标是 LabVIEW 语言很有特色的一个东西，其它语言大概都不需要给它的函数设置一个图形标识。认真给每一个 VI 加上一个有意义的图标是非常必要的，它可以帮助程序人员快速理解一段程序的含义。一个风格良好的 VI，只要扫一眼它的程序框图，就可以了解他的功能了，而不必要像文本语言那样一行行把程序读下来。LabVIEW 程序比文本语言的程序可读性要高得多，当然这是指在编程风格良好的情况下进行比较。
有意义的图标是良好程序风格中非常重要的一项。编辑图标应该是一个很常见的工作，双击一个 VI 的图标就会出现图标编辑器，如下图所示。



上图这个这个图标编辑器界面左上方是绘制图标的工具栏，里面工具被我编了号。不知道大家有没有试过鼠标双击这几个工具，我是经常双击他们的。

编辑一个 VI 的图标，第一件事就是把他的默认图标删除，留一个空白框在上面就可以了。做这件工作不需要选中图标中的内容再删除，只要双击一次C2这个工具就可以了。
如果不需要清楚图标中的内容，只想给他添个边框，那就双击C1这个工具。
如果想选中整个图标，以备拷贝，按 Ctrl+A 是没用的，可以双击D1工具。
双击D2，会出现一个字体设置对话框，在这里改变写到图标上的字体。如果要在图标上写英文，最适合的字体是“Small Fonts”，字号8。

**Reply by ruanqizhen** [reply]
之前博客下的留言：

Qizhen
2009年03月11日 17:31 编辑
这个是我自己发现的，不过后来看了LabVIEW在线帮助，上面也有介绍。

回复
帆
2009年02月25日 16:20 编辑
这个技巧在哪里发现的？自己摸索还是有文档记载？

回复
刘伟印
2008年08月6日 13:12 编辑
不错，作者很细心啊

回复
agilentvee
2008年01月2日 17:05 编辑
Small Fonts”，字号8。好象太小,最好是Arial 12号字体

回复
竹贤
2007年12月27日 20:07 编辑
尽管是小技巧，但以前从未知道。随手用来，确实方便。希望能够多介绍一些，我们也可有所收益。

**Comment by ruanqizhen** [comment]
之前博客上的相关文章，搬到这里：

## 制作不规则图形的子VI图标

大多数子 VI 的图标都是一个正方块。但有时候，为了程序代码美观、易读，需要把子 VI 作成不规则图形显示在调用它的父 VI 代码上。LabVIEW 中很多自带的函数采用的就不是32×32的正方形，比如说加减乘除运算函数等。我们已可以把子 VI 作成非正方形方块，比如图1中的 My Function.vi，我把它的图标做成了三角形。


图1：调用 My Function.vi 的代码

把子 VI 的图标改为非规则图形，有一些事项需要注意。首先要考虑选取一个合适的接线方块模式。现在 LabVIEW 默认的模式是4224模式（按照每一列接线端的数量给模式命名的）。但这种模式通常不适合不规则的图标。比如说图1的 My Function.vi，它需要一个纵向排在中央的输出接线端，4224模式最右侧的4个接线端对称分布，没有一个在最中间。
在VI前面板的图标上点击鼠标右键，选择“Show Connector”，之后再鼠标右击，就可以挑选一个合适的模式了。对于 My Function .vi，可以使用52225这个模式。如下图所示：

   
图2、3：选取接线方块的模式

接下来，就是核心工作-编辑图标了。制作不规则形状的图标最重要的两点，一是不要给图标加上边框，二是三个颜色的图标要保持一致。下图是 LabVIEW 的图标编辑器，它的中央有三个小方块，分别显示黑白色、16色、256色下的VI图标。用鼠标点击一下某个小方块，左边的编辑区就开始对这个颜色的图标进行编辑。
如果你的显示器是颜色数量高于256色，那么，你在程序代码中看到的始终是256色那个图标；如果少于这个颜色数量，比如只有16色，那么在程序代码上看的的子VI显示的就是16色的那个图标。不过现在恐怕找不到16色的显示器了，所以16色这个图标一般情况下就是空白。黑白色的显示器虽然更罕见，但是这里的黑白图标还有另一个用处，就是确定图标中图形的轮廓。所以这里一定要有一个形状和256色图标项符合的黑白图标才能做出不规则图形图标来。一般在编辑黑白图标时，按一下“Copy from 256 Colors”的按钮，把256色的图标复制过来就可以了。
编辑不规则形状的图标最好选中“Show Terminals”，把接线方块模式在图标编辑区显示出来。这样可以方便的查看，现在的图形是否与接线方块般配。比如说，我要在接线方块右侧中央的接线端接一个输出数据，我的图形至少要覆盖到这个接线端才行。

**Reply by ruanqizhen** [reply]
之前博客下的留言：

ruanqizhen
2011年03月23日 10:50 编辑
我对你问的问题没有什么研究，不过听起来你的问题更像是曲线吧。
如果你的图形很复杂，可以试试看用LabVIEW的3D picture control画。

回复
wsmaggie512
2011年03月22日 11:17 编辑
hello ,i want to ask you some problem about of labview, please instruct a guide,thank you !
我想用labview画个立体图，在xy平面上一条曲线，表示的是船走的航线，在船停留的那一处，z轴上也有一相应数据，表示的是水深，我想在画航线的同时，也能画出相应的水深，而在航线曲线上，能否画出切线代表水的流速？流速的大小可以使航行线有宽有窄。这个用labview的三维曲面还是三维曲线画比较好画？用曲线的话，显示控件上能否同时显示面和线？曲线之间能否进行颜色填充，让画面更直观一些？还是用matlab画图，然后用labview调用更简单一些？很多问题网上查不到相应的资料，希望您能给我指点一下，非常感谢~！

回复

**Reply by ruanqizhen** [reply]
之前博客下的留言：

志恒
2010年01月22日 20:16 编辑
新版图标编辑器的图标全部以png格式存在我的文档的“LabVIEW Data”中，使用新版图标编辑器的时候需要调用，一旦隐藏后调用失败就会启动旧版。

回复
志恒
2010年01月22日 20:10 编辑
我发现了一个更简单的禁用方法。把我的文档下的“LabVIEW Data”文件夹隐藏后，打开“图标编辑器”以后新版的图标编辑器一闪而过，然后出现旧版。如果要用新版的话取消文件夹隐藏就可以了。

回复
l
2009年12月31日 17:02 编辑
我已经写好了一篇blog并且发送了引用链接,你可以在下面的Trackback看到

回复
Qizhen
2009年12月31日 11:21 编辑
Cleartype在哪里设置，我还真不知道，能详细说一下吗？我用开源这个词可能不太好，严谨的说应该是“源代码公开的” :)

回复
l
2009年12月31日 09:56 编辑
字体模糊是因为打开了Cleartype的缘故,禁用就可以了.另外,开源不只是能查看源代码,还和授权协议有关,这个图标编辑器的授权应该是包含在LabVIEW里面的,是否能算作开源值得商榷.

回复
Qizhen
2009年12月30日 11:01 编辑
禁用新图标编辑器的方法是 Billy Wei 告诉我的，在此表示感谢 :)

回复
